src/commands/test/examples/cheats/test_prank.cairo

```diff
@@ -73,6 +73,39 @@ func test_syscall_counter_correct{syscall_ptr : felt*, range_check_ptr}():
     %}
     let (caller_addr) = get_caller_address()
     assert caller_addr = 345
+    # We check if syscall counter has been correctly incremented
     let (bn) = get_block_number()
     return ()
```

Comment:
There's no assertion for that. Does this function throw an error if syscall_counter wasn't correctly incremented?

Reply:
Yes, if you call a syscall and the previous syscall won't increment the syscall counter, it will throw an error. We had this bug before.

Reply:
Added more comment about it